src/test/scala/edu/ie3/simona/model/participant/BMModelSpec.scala

```diff
@@ -0,0 +1,338 @@
+        0.61d,
+      ), // Day 150 of the year
+      (
+        "2019-08-31T05:15:00+02:00[Europe/Berlin]",
+        0.61d,
+      ), // Day 243 of the year
+      ("2019-09-01T05:15:00+02:00[Europe/Berlin]", 1.03d), // Day 244 of the year
+    )
+
+    testCases.foreach { case (time, k2Sol) =>
+      val k2Calc = bmModel.calculateK2(ZonedDateTime.parse(time))
+      k2Calc should be(k2Sol)
+    }
+  }
+
+  "calculate PTh correctly" in {
+    val bmModel = buildBmModel()
+
+    val testCases = Table(
+      ("Temperature", "K1", "K2", "PTh Sol"),
+      (19.28, 1d, 1d, 5.62d), // independent of temp
+      (30d, 2d, 3d, 33.72d),
+      (19.2799999d, 1d, 1d, 5.6147201076d), // dependent on temp
+      (15d, 1.01d, 0.61d, 6.296542d), // somewhat realistic
+    )
+
+    testCases.foreach { case (temp, k1, k2, pThSol) =>
+      val pThCalc = bmModel.calculatePTh(Celsius(temp), k1, k2)
+      pThCalc should approximate(Megawatts(pThSol))(powerTolerance)
+    }
+  }
+
+  "calculate usage correctly" in {
+    val bmModel = buildBmModel()
+
+    val testCases = Table(
+      ("PTh", "Usage Solution"),
+      (43.14d, 1d), // exactly maximum heat
+      (50d, 1d), // more than maximum, cap to 1
+      (20d, 0.463606861382d), // less than max
+      (0d, 0d), // zero
+    )
+
+    testCases.foreach { case (pTh, usageSol) =>
+      val usageCalc = bmModel.calculateUsage(Megawatts(pTh))
+      usageCalc should be(usageSol +- usageTolerance)
+    }
+  }
+
+  "calculate efficiency correctly" in {
+    val bmModel = buildBmModel()
+
+    val testCases = Table(
+      ("Usage", "Efficiency Sol"),
+      (1d, 1d),
+      (0d, 0.724d),
+      (0.75d, 0.98425d),
+      (0.86446317d, 0.993848918615d),
+    )
+
+    testCases.foreach { case (usage, effSol) =>
+      val effCalc = bmModel.calculateEff(usage)
+      effCalc should be(effSol +- 0.000000001)
+    }
+  }
+
+  "calculate electrical output correctly" in {
+
+    val testCases = Table(
+      ("FeedInTariff", "Usage", "Efficiency", "PEl Sol"),
+      (0.051d, 1d, 1d, -190d), // tariff greater than opex => full power
+      (
+        0.04d,
+        0.75d,
+        0.98425d,
+        -140.255625d,
+      ), // tariff too little, only serve heat demand
+      (0.04d, 1d, 1d, -190d), // tariff too little, but max heat demand
+    )
+
+    testCases.foreach { case (feedInTariff, usage, eff, pElSol) =>
+      val bmModel = new BMModel(
+        UUID.fromString("8fbaf82d-5170-4636-bd7a-790eccbea880"),
+        "BM Model Test",
+        OperationInterval(0L, 86400L),
+        QControl(new CosPhiFixed("cosPhiFixed:{(0.0,1.0)}")),
+        Kilowatts(190),
+        bmType.getCosPhiRated,
+        "MockNode",
+        isCostControlled = true,
+        EUR(bmType.getOpex.getValue.doubleValue()),
+        EuroPerKilowatthour(feedInTariff),
+        0.05,
+      )
+
+      val pElCalc = bmModel.calculateElOutput(usage, eff)
+      pElCalc.value should be(Kilowatts(pElSol).value +- 1e-4)
```

Comment:
If you remove the value and the tolerance this should be tested for the correct unit as well